src/ffs/ffs.ts

```diff
@@ -0,0 +1,104 @@
+import pubTree, { PublicTree } from './pub/tree'
+import privTree, { PrivateTree } from './priv/tree'
+import pubNode, { PublicNode } from './pub/node'
+import { Tree, Node, Link } from './types'
+import { CID, FileContent } from '../ipfs'
+import pathUtil from './path'
+
+export class FFS {
+
+  root: PublicNode
+  pubTree: PublicTree
+  privTree: PrivateTree
```

Comment:
More from the conversation in Discord:

If we rearrange stuff, this could be perhaps streamlined and the stateful classes split out (functional OO design). For example:

```ts
class Session {
  // Stateful
  // Manages where the user "is" in the system, mutable updates, and so on
  // Think of this as a bash session
}

class Directory {
  // Stateless
  // i.e. Currently `PubTree`

  cid: CID
  files: {String => File}
  directories: {String => Directory}

  // ...
}

class Directory.Private 
  extends Directory 
  implements Private {
    key: AES256
    // ...
}

class File {
  cid: CID
  data: Buffer // or something

  constructor(cid_: CID) {
    this.cid = cid_
  }

  // ...some methods
}

class File.Private 
  extends File
  implements Private {
    key: AES256

    // ...
}
```